packages/dev/src/index.js

```diff
@@ -118,34 +144,43 @@ class Server {
     let booted = false
     const boot = promiseDefer()
     this.enqueue(() => boot.promise)
-    this.subscriptions.add(await this.pundle.loadComponents([
-      [cliReporter, {
-        log: (text, error) => {
-          if (this.config.hmrReports && error.severity && error.severity !== 'info') {
-            this.writeToConnections({ type: 'report', text, severity: error.severity || 'error' })
-          }
-        },
-      }],
-      createWatcher({
-        tick: (_: Object, __: Object, file: File) => {
-          if (booted && file.filePath !== Helpers.browserFile) {
-            this.state.changed.set(file.filePath, file)
-          }
-        },
-        ready: () => {
-          this.report('Server initialized successfully')
-        },
-        compile: async (_: Object, __: Object, chunks: Array<FileChunk>, files: Map<string, File>) => {
-          this.state.files = files
-          this.state.chunks = chunks
-          if (this.connections.size && this.state.changed.size) {
-            await this.generateForHMR()
-          }
-          boot.resolve()
-          booted = true
-        },
-      }),
-    ]))
+    this.subscriptions.add(
+      await this.pundle.loadComponents([
+        [
+          cliReporter,
+          {
+            log: (text, error) => {
+              if (this.config.hmrReports && error.severity && error.severity !== 'info') {
+                this.writeToConnections({ type: 'report', text, severity: error.severity || 'error' })
+              }
+            },
+          },
+        ],
+        createWatcher({
+          tick: (_: Object, __: Object, file: File) => {
+            if (booted && file.filePath !== Helpers.browserFile) {
+              this.state.changed.set(file.filePath, file)
+              this.state.hasChanged = true
```

Comment:
Yep, I believe I had tried that but it wasn't working for whatever reason. I can try again.